docs/guide/topics/webhooks.rst

```diff
@@ -0,0 +1,219 @@
+working with them using discord.py.
+
+For detailed information about webhooks, See the documentation for :ref:`webhooks <discord-api-webhooks>`.
+
+Introduction to webhooks
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The difference between sending messages with webhooks and bots is that bots require proper authentication and a bot user
+whereas webhooks don't need either.
+
+There are currently three types of webhooks:
+
+1. Incoming webhooks
+2. Channel follower webhooks
+3. Application webhooks
+
+**Incoming webhooks** are associated to a channel and can post messages in that channel
+using a webhook token.
+
+**Channel follower webhooks** are used for crossposting of messages in news channels
+by Discord. This type does not have any token associated to it and bots cannot post
+messages through these webhooks.
+
+**Application webhooks** are used for creating interaction responses. The management of
+this type of webhooks are handled internally by the library and is not in scope of this guide.
+
+This guide will primarily focus on the first type i.e Incoming webhooks.
+
+Creating a webhook
+~~~~~~~~~~~~~~~~~~
+
+In order to create incoming webhooks in a channel, the member needs the
+:attr:`~Permissions.manage_webhooks` permission in the given guild or text channel.
+
+To create a webhook, Navigate to "Server Settings > Integrations" page.
+
+.. image:: /images/guide/webhooks/guild_settings_integrations.png
+    :alt: Integrations Settings
+
+Click "Create Webhook" to create a webhook.
+
+.. image:: /images/guide/webhooks/webhook_created.png
+    :alt: Webhook
+
+From here, you can edit the webhook's name and the associated channel the webhook
+will send messages in. We'll be using the URL of webhook later in this guide so make
+sure to copy it too.
+
+The URL of webhook is used for executing the webhook and includes the webhook token
+so don't store it in a public place.
+
+You can also create webhooks through API using the :meth:`TextChannel.create_webhook` method.
+
+Initializing webhooks
+~~~~~~~~~~~~~~~~~~~~~
+
+We'll now be initializing a webhook instance in our code.
+
+There are two variants of webhooks provided by the library.
+
+- :class:`Webhook`
+- :class:`SyncWebhook`
+
+The :class:`Webhook` is suitable for most cases as it is asynchronous while
+:class:`SyncWebhook` is synchronous and is suitable for blocking environments. Both have the same
+functionality with the difference of being sync and async. The HTTP methods usually return
+the :class:`Webhook` class instance.
+
+Webhooks are retrieved in two ways. They can be obtained through the :meth:`Guild.webhooks`
+or :meth:`TextChannel.webhooks` methods. These methods requires authentication via a bot user.
+
+The other method is to initialize "partial" webhooks using only webhook ID and token. This
+method does not require authentication via bot user.
+
+To create a partial webhook we would be using the URL that we copied earlier, passing it
+in the :meth:`~Webhook.from_url` method.
```

Comment:
Do we intend to mention `Client.fetch_webhook` either here or in the section at L110?

Reply:
The latter seems a more suitable place for it. I have mentioned it there. Suggestions for wording would be appreciated. :>